### ISSUE TITLE
Corrupted fcall_info provided to zend_call_function() in Unknown on line 0

### ISSUE THREAD

**Issue by aizuyan** [php-zookeeper/php-zookeeper#28]
I want to implement a lock lib but trigger error
```
开始获取锁
PHP Notice:  Undefined property: ZkLock::$watching in /Users/ritoyan/imoocCode/locks/test/ZkLock.php on line 137

Notice: Undefined property: ZkLock::$watching in /Users/ritoyan/imoocCode/locks/test/ZkLock.php on line 137
PHP Fatal error:  Corrupted fcall_info provided to zend_call_function() in Unknown on line 0

Fatal error: Corrupted fcall_info provided to zend_call_function() in Unknown on line 0
```

maybe called by the var `watching` in callback

the fllow is the source code 

```php
<?php
/****************************************************************
 *                                                              *
 * Copyright (c) 2018 zuoyebang.com, Inc. All Rights Reserved.  *
 *                                                              *
 ****************************************************************
 * Created by PhpStorm.
 * User: ritoyan
 * Date: 2018/10/5
 * Time: 下午12:34
 */
class ZkLock extends Zookeeper
{
    /**
     * 锁序号文件的前缀
     *
     * @var string
     */
    public $lockPrefix = "lock-";

    /**
     * 获取失败的时候，需要监控的节点
     *
     * @var string
     */
    protected $watchPath = "";

    /**
     * 是否监控标识，循环依靠这个停止
     *
     * @var bool
     */
    public $watching = false;

    /**
     * 根据nodepath，解析node的序号
     *
     * @param $nodePath
     * @return int
     */
    protected function parseSeq($nodePath)
    {
        $parts = explode("-", $nodePath);
        return intval($parts[1]);
    }

    /**
     * 获取锁
     *
     * @param $path 锁序号存放的目录
     * @param $acl
     * @return bool
     */
    public function lock($path, $acl)
    {
        $path       = rtrim($path, "/");
        $createPath =  $path . "/" . $this->lockPrefix;

        for ($i = 0; $i < 3; $i++) {
            try {
                $lockNode = $this->create($createPath, "", $acl, Zookeeper::EPHEMERAL | Zookeeper::SEQUENCE);

                if ($lockNode) {
                    break;
                }
            } catch (ZookeeperNoNodeException $e) {
                $paths      = explode("/", trim($path, "/"));
                $nodePath   = "";
                foreach ($paths as $path) {
                    $nodePath .= "/" . $path;
                    try {
                        $ret = $this->create($nodePath, "", $acl);
                        if ($ret) {
                            continue;
                        }
                    } catch (Exception $e) {
                        break;
                    }
                    break;
                }
            } catch (Exception $e) {
                break;
            }

            return false;
        }

        $mySeq          = $this->parseSeq($lockNode);
        while (true) {
            $childNodes = $this->getChildren($path);
            $lowestSeq = $mySeq;
            $preMySeq = -1;
            $preNodePath = "";
            foreach ($childNodes as $childNode) {
                $seq = $this->parseSeq($childNode);
                if ($seq < $lowestSeq) {
                    $lowestSeq = $seq;
                }

                if ($mySeq > $seq && $seq > $preMySeq) {
                    $preMySeq = $seq;
                    $preNodePath = $childNode;
                }
            }

            if ($mySeq == $lowestSeq) {
                // 获取锁成功
                return true;
            } else {
                // 需要监听比我序号小的节点
                $watchNodePath = $path . "/" . $preNodePath;
                try {
                    $this->watch($watchNodePath);
                    continue;
                } catch (ZookeeperNoNodeException $e) {
                    var_dump(123);
                    continue;
                } catch (Exception $e) {
                    return false;
                }
            }
        }
    }

    /**
     * 监控某个节点的删除事件
     *
     * @param $nodePath
     * @return bool
     */
    public function watch($nodePath)
    {
        $this->watchPath    = $nodePath;
        $this->watching     = true;
        $ret                = $this->exists($nodePath, [$this, "watcher"]);
        while (true) {
            if (!$this->watching) {
                break;
            }
            sleep(2);
        }
        return true;
    }

    /**
     * 具体的事件触发毁掉函数
     *
     * @param $eventType int 事件类型
     * @param $stat int
     * @param $path string
     */
    public function watcher($eventType, $stat, $path)
    {
        var_dump(123);
        if (self::DELETED_EVENT == $eventType) {
            $this->watching = false;
        } else {
            $this->exists($this->watchPath, [$this, "watcher"]);
        }
    }
}


$zk = new ZkLock("localhost:2181");
echo "开始获取锁\n";
$r = $zk->lock("/yrtyyy/names", [
    [
        'perms'  => Zookeeper::PERM_ALL,
        'scheme' => 'world',
        'id'     => 'anyone',
    ]
]);

if ($r) {
    echo "获取成功\n";
} else {
    echo "获取失败\n";
}

sleep(5);
var_dump("do something\n");
```

**Comment by aizuyan** [comment]
env info:

```
➜  test php -v
PHP 7.2.10 (cli) (built: Sep 14 2018 07:07:08) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.10, Copyright (c) 1999-2018, by Zend Technologies
```

```
➜  locks php -i | grep --color -rn zookeeper
(standard input):779:Registered save handlers => files user zookeeper memcached redis rediscluster
(standard input):1014:zookeeper
(standard input):1016:zookeeper support => enabled
(standard input):1018:libzookeeper version => 3.4.9
(standard input):1021:zookeeper.recv_timeout => 10000 => 10000
(standard input):1022:zookeeper.sess_lock_wait => 150000 => 150000
(standard input):1023:zookeeper.session_lock => 1 => 1
(standard input):1039:PWD => /Users/ritoyan/Downloads/php-zookeeper-0.4.0
(standard input):1075:$_SERVER['PWD'] => /Users/ritoyan/Downloads/php-zookeeper-0.4.0
```

**Comment by Timandes** [comment]
Would you try newer version of php-zookeeper? 
We fixed a issue in `0.4.1`. It's pretty much like what you reported.
